### ISSUE TITLE
[Bug] Build failing

### ISSUE THREAD

**Issue by Aeyos** [mxstbr/react-boilerplate#1475]
Build fails on webpack's UglifyJS, like so: 

```
ERROR in main.787b352c844d558cf92f.js from UglifyJs
SyntaxError: Unexpected token: punc (:) [main.787b352c844d558cf92f.js:39193,15]
```

The segment of code looks like this
```
function save(namespaces) {
  if (null == namespaces) {
    // If you set a process.env field to null or undefined, it gets cast to the
    // string 'null' or 'undefined'. Just delete instead.
    delete {"NODE_ENV":"production"}.DEBUG;
  } else {
    {"NODE_ENV":"production"}.DEBUG = namespaces;
  }
}
```

I figured out there must be a conflict with the environment variables somewhere or a module included by mistake (debug), which is the node's implementation of it. I tried my best to narrow down the error, but this is beyond my current capabilities. I failed at trying to exclude this module out of the building process.

Could someone please shed some light in this issue? I would be most grateful!

## Steps to reproduce

```
$ npm run build
```

**Comment by Aeyos** [comment]
There was two things that kept me from successfully building the code.

FS and NET modules were being used by some package in the client-side
```
node: {
  fs: 'empty',
  net: 'empty',
},
```

And the debug module was also being included in the bundle. After aimlessly wandering through stack overflow threads I found out the problem and how to fix it simply by adding:
```
externals: ['debug'],
```

I'm sorry if I caused any troubles, I was getting desperate for an answer.
